docs/reference/troubleshooting/common-issues/hotspotting.asciidoc

```diff
@@ -0,0 +1,265 @@
+
+Hot spotting most commonly surfaces as significantly elevated 
+resource utilization (of `disk.percent`, `heap.percent`, or `cpu`) among a 
+subset of nodes as reported via <<cat-nodes,cat nodes>>. Individual spikes aren't 
+necessarily problematic, but if utilization repeatedly spikes or consistently remains 
+high over time (e.g. `>30s`) then that resource may be experiencing problematic 
+hot spotting. 
+
+For example, let's show case two separate plausible issues using cat nodes:
+
+[source,console]
+----
+GET _cat/nodes?v&s=master,name&h=name,master,node.role,heap.percent,disk.used_percent,cpu
+----
+Pretend this same output pulled twice across five minutes:
+
+[source,console-result]
+----
+name   master node.role heap.percent disk.used_percent cpu
+node_1 *      hirstm              24                20  95
+node_2 -      hirstm              23                18  18
+node_3 -      hirstmv             25                90  10
+----
+// TEST[skip:illustrative response only]
+
+Here we see two significantly unique utilizations: where the master node is at 
+`cpu: 95` and a hot node is at `disk.used_percent: 90%`. This would indicate 
+hot spotting was occurring on these two nodes, and not necessarily from the same
+root cause. 
+
+[discrete]
+[[causes]]
+==== Causes
+
+Historically, clusters experience hot spotting mainly as an effect of hardware, 
+shard distributions, and/or task load. We'll review these sequentially in order 
+of their potentially impacting scope.
+
+[discrete]
+[[causes-hardware]]
+===== Hardware
+
+Here are some common improper hardware setups which may contribute to hot 
+spotting:
+
+* Resources are allocated non-uniformly. For example, if one hot node is 
+given half the cpu of its peers. {es} expects all nodes on a 
+<<data-tiers,data tier>> to share the same hardware profiles or 
+specifications.
+
+* Resources are consumed by another service on the host, including other 
+{es} nodes. Refer to our <<dedicated-host,dedicated host>> recommendation.
+
+* Resources experience different network or disk throughputs. For example, if one 
+node's I/O is lower than its peers. Refer to 
+<<tune-for-indexing-speed,Use faster hardware>> for more information.
+
+* A JVM that has been configured with a heap larger than 31GB. Refer to <<set-jvm-heap-size>> 
+for more information.
+
+* Problematic resources uniquely report <<setup-configuration-memory,memory swapping>>. 
+
+[discrete]
+[[causes-shards]]
+===== Shard distributions
+
+{es} indices are divided into one or more link:{wikipedia}/Shard_(database_architecture)[shards] 
+which can sometimes be poorly distributed. {es} accounts for this by <<modules-cluster,balancing shard counts>> 
+across data nodes. As link:{blog-ref}whats-new-elasticsearch-kibana-cloud-8-6-0[introduced in version 8.6], 
+{es} by default also enables <<modules-cluster,desired balancing>> to account for ingest load; 
+though there's always link:{es-repo}/issues/17213[room for improvement]. 
```

Comment:
```suggestion
{es} by default also enables <<modules-cluster,desired balancing>> to account for ingest load.. 
```

Reply:
Is the "room for improvement" note helpful?

Reply:
It links to the parent discussion about why shard hot spotting occurs at all in Elasticsearch vs our attempts to account for it. I think it's helpful for advanced users, but maybe not newer. What do you think?

Reply:
I'm not sure it belongs in documentation. A Github conversation is ephemeral, and who knows if it will still be relevant a year from now, for example. But if you feel strongly, we can keep it, of course.